comp/logs/internal/sourcemgr/component.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package sourcemgr implements a component managing logs-agent sources (type
+// LogSource).  It receives additions and removals of sources from other
+// components, and it informs subscribers of these additions and removals.
+//
+// Subscribers should provide a Subscription value in value-group "sourcemgr".
+//
+// Once added to this component, a LogSource must be considered immutable: neither
+// the component having called AddSource, nor any of the subscribers, may modify the
+// source.
+//
+// All component methods can be called concurrently.
+package sourcemgr
+
+import (
+	"github.com/DataDog/dd-agent-comp-experiments/pkg/util/subscriptions"
+	"go.uber.org/fx"
+)
+
+// team: agent-metrics-logs
+
+const componentName = "comp/logs/internal/sourcemgr"
+
+// Component is the component type.
+type Component interface {
+	// AddSource adds a new log source.
+	AddSource(*LogSource)
+
+	// RemoveSource removes an existing log source.
+	RemoveSource(*LogSource)
+}
+
+// LogSource defines a source for logs.
+type LogSource struct {
+	Name string
+}
+
+// Launcher defines the interface each launcher must satisfy.
+type SourceChange struct {
+	// IsAdd is true when the source is being added, and false when it is being removed.
+	IsAdd bool
```

Comment:
The idea is "is this SourceChange an add", so IsAdded might not scan well.

WLM [uses a type and some consts](https://github.com/DataDog/datadog-agent/blob/e341951a44f8a7491818c91a371a7a0ecf01c847/pkg/workloadmeta/types.go#L601-L605) so maybe it would be better to follow that pattern.